Fails to compile on linux aarch64

I have a mac with Apple M1 Pro CPU and I use a docker with ubuntu:20.04 image. When I include `numpy = "0.16"` as my dependency, it fails to compile with the following error:

```
   Compiling numpy v0.16.1
error[E0600]: cannot apply unary operator `-` to type `u8`
  --> /home/dev/.cargo/registry/src/github.com-1ecc6299db9ec823/numpy-0.16.1/src/npyffi/flags.rs:74:42
   |
74 | pub const NPY_ALIGNED_STRUCT: npy_char = -128; // 0x80
   |                                          ^^^^ cannot apply unary operator `-`
   |
   = note: unsigned values cannot be negated

For more information about this error, try `rustc --explain E0600`.
error: could not compile `numpy` due to previous error
```

From the [documentation and source code](https://doc.rust-lang.org/std/os/raw/type.c_char.html), it appears npy_char, which is c_char, will be `u8`. 

I am not familiar with the artitecture of rust-numpy to offer a solution or PR. Could anyone suggest a quick fix for this issue?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 # Changelog
 
 - Unreleased
+  - Fix build on platforms where `c_char` is `u8` like Linux/AArch64. ([#296](https://github.com/PyO3/rust-numpy/pull/296))
 
 - v0.16.1
   - Fix build when PyO3's `multiple-pymethods` feature is used. ([#288](https://github.com/PyO3/rust-numpy/pull/288))

--- a/src/npyffi/flags.rs
+++ b/src/npyffi/flags.rs
@@ -71,7 +71,8 @@ pub const NPY_NEEDS_INIT: npy_char = 0x08;
 pub const NPY_NEEDS_PYAPI: npy_char = 0x10;
 pub const NPY_USE_GETITEM: npy_char = 0x20;
 pub const NPY_USE_SETITEM: npy_char = 0x40;
-pub const NPY_ALIGNED_STRUCT: npy_char = -128; // 0x80
+#[allow(overflowing_literals)]
+pub const NPY_ALIGNED_STRUCT: npy_char = 0x80;
 pub const NPY_FROM_FIELDS: npy_char =
     NPY_NEEDS_INIT | NPY_LIST_PICKLE | NPY_ITEM_REFCOUNT | NPY_NEEDS_PYAPI;
 pub const NPY_OBJECT_DTYPE_FLAGS: npy_char = NPY_LIST_PICKLE

